src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/DirectiveAttributeCompletionItemProvider.cs

```diff
@@ -77,7 +77,15 @@ public override IReadOnlyList<RazorCompletionItem> GetCompletionItems(RazorSynta
 
             var completionItems = GetAttributeCompletions(attributeName, containingTagName, attributes, tagHelperDocumentContext);
 
-            return completionItems;
+            if (string.IsNullOrWhiteSpace(attributeName))
+            {
+                return completionItems;
+            }
+
+            // We filter out the completions offered to match the partially completed attribute in context.
+            // Ie. don't offer completions part way through an existing unrelated attribute.
+            // <svg xml:| ></svg> (attributeName = "xml:") should not get any directive attribute completions.
+            return completionItems.Where(c => c.DisplayText.StartsWith(attributeName)).ToList();
```

Comment:
This seems a little bit too restrictive. I'd say if the attribute starts with `@`  then it's potentially a directive attribute. That way you can swap between directive attributes

Reply:
Hmm, fair point. Though I'm not sure how much swapping may be possible after VS filters out the completions later in the pipeline. Making the change 😄

Reply:
> Though I'm not sure how much swapping may be possible after VS filters out the completions later in the pipeline.

In VSCode once a completion session starts I don't believe it "refreshes" it until the previous session was completed. So if you don't have the items already available then changing the attribute name via deleting some content wouldn't result in those completions being shown.

Reply:
Makes sense, thanks. Updated 👍